### ISSUE TITLE
Too Many False Positives On Node.js Projects

### ISSUE THREAD

**Issue by MarcL** [dependency-check/DependencyCheck#495]
Hi Jeremy,

I've been using the `dependency-check` command line tools on some Node.js projects and found that I was getting a lot of false positives. These were mainly relating to other languages or frameworks (e.g. Drupal, PHP or C code rather than the JavaScript code from the dependent Node packages).

Looking at the Node analyser [documentation](http://jeremylong.github.io/DependencyCheck/analyzers/nodejs.html), it seems like evidence is gathered by this but is then subsequently incorrectly classified via subsequent analysers and creates CPEs for other languages. The confidence level for all of these is low, and I could just add them to the suppression file. For a project with 13 potential vulnerabilities, only 4 of them were Node.js specific and all of those were relating to an older version of Node (0.10.1) than we are using (4.2.2 I think).

I've created a simple Node.js project for testing (should you want it). It contains one simple dependent package which was picked up in our full projects. The test repo is here:

https://github.com/MarcL/dependency-check-test

and the dependency report can be seen here:

https://htmlpreview.github.io/?https://github.com/MarcL/dependency-check-test/blob/master/report/dependency-check-report.html

I'm not sure if this is classed as a bug or just a limitation of the dependency-check tool for Node.js projects. Is there a better way to fix this than suppressing all of the false positives? Can/should the evidence be tagged by project type to avoid this false classification?

Thanks,
Marc

**Comment by jeremylong** [comment]
The issue is that the data being used (NVD) does not tag vulnerabilities as being to a specific language. I have spent a bit of time on the Java side creating suppression rules to reduce the issue you are seeing in Node.js. See examples in the [base suppression xml](https://github.com/jeremylong/DependencyCheck/blob/master/dependency-check-core/src/main/resources/dependencycheck-base-suppression.xml#L37). A large volume of these false positives can be taken care of quickly by creating similar entries (PRs are more then welcome).

I know node.js was added to dependency-check - but for many reasons using tools like [retire.js](http://retirejs.github.io/retire.js/) or [node security project](https://nodesecurity.io/tools) are likely going to be much more accurate.

Due to the false positive rates associated with a  few a the analyzers I am going to flag them as  experimental and require specifically turning on the experimental flag. See issue #498.
